homeassistant/components/lifx/coordinator.py

```diff
@@ -189,41 +192,54 @@ def async_get_entity_id(self, platform: Platform, key: str) -> str | None:
     async def _async_update_data(self) -> None:
         """Fetch all device data from the api."""
         async with self.lock:
```

Comment:
I would remove this lock, btw. I added it in the previous integration's incarnation to prevent multiple discoveries from firing at the same time, but that's no longer an issue (or possible). 

For reference, Photons uses a Semaphore instead of a lock to limit the number of packets "in-flight", i.e. awaiting a response, but that's not bulb specific. Photons is essentially just a very fancy way of getting the least amount of packets on the network to do the most amount of work across the largest number of devices.